src/openloco/windows/station_list.cpp

```diff
@@ -17,36 +37,738 @@ namespace openloco::ui::windows::station_list
+        if (widgetIndex == widx::company_select)
+            dropdown::populateCompanySelect(window, &window->widgets[widgetIndex]);
+    }
+
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::close_button:
+                WindowManager::close(window);
+                break;
+
+            case tab_all_stations:
+            case tab_rail_stations:
+            case tab_road_stations:
+            case tab_airports:
+            case tab_ship_ports:
+            {
+                if (input::is_tool_active(window->type, window->number))
+                    input::cancel_tool();
+
+                window->current_tab = widgetIndex - 4;
+                window->frame_no = 0;
+
+                window->invalidate();
+
+                window->var_83C = 0;
+                window->row_hover = -1;
+
+                sub_4910E8(window);
+
+                window->call_on_resize();
+                window->call_prepare_draw();
+                window->init_scroll_widgets();
+
+                registers regs;
+                regs.esi = (int32_t)window;
+                regs.edx = widgetIndex;
+                call(0x004CD320, regs);
+                break;
+            }
+
+            case sort_name:
+            case sort_status:
+            case sort_total_waiting:
+            case sort_accepts:
+            {
+                auto sort_mode = widgetIndex - 10;
+                if (window->var_844 == sort_mode)
+                    return;
+
+                window->var_844 = sort_mode;
+                window->invalidate();
+                window->var_83C = 0;
+                window->row_hover = -1;
+
+                sub_4910E8(window);
+                break;
+            }
+        }
+    }
+
+    // 0x00491A0C
+    static void on_scroll_mouse_down(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        uint16_t currentRow = y / rowHeight;
+        if (currentRow > window->var_83C)
+            return;
+
+        int16_t currentStation = window->row_info[currentRow];
+        if (currentStation == -1)
+            return;
+
+        windows::station::open(currentStation);
+    }
+
+    // 0x004919D1
+    static void on_scroll_mouse_over(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        window->flags &= ~(window_flags::flag_14);
+
+        uint16_t currentRow = y / rowHeight;
+        uint16_t currentStation = 0xFFFF;
+
+        if (currentRow < window->var_83C)
+            currentStation = window->row_info[currentRow];
+
+        if (currentStation == window->row_hover)
+            return;
+
+        window->row_hover = currentStation;
+        window->invalidate();
+    }
+
+    // 0x0049193F
+    static void on_update(window* window)
+    {
+        window->frame_no++;
+
+        window->call_prepare_draw();
+        WindowManager::invalidate(WindowType::stationList, window->number);
```

Comment:
WindowManager::invalidateWidget(WindowType::stationList, window->number, window->current_tab + 4);
```suggestion
        WindowManager::invalidateWidget(WindowType::stationList, window->number, window->current_tab + 4);
```